src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -33,21 +36,58 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     UserService userService = UserServiceFactory.getUserService();
-
+    
+    // Set default variables to create UserAuthenticationData object.
     Boolean loggedIn = false;
+    String email = "";
+    Boolean isUserRegistered = false;
     String authenticationUrl = "";
-    String userEmail = "";
     String redirectUrl = "/";
+
+    // If user is logged in, udpate variables.
+    // Set authenticationUrl to either logout or login URL.
```

Comment:
In this case, I'm not sure if these comments are really adding any additional clarity at the top level, it's already clear that if a user is logged in, we update variables. You could move the authenticationUrl comment down closer to where it is actually happening.

Reply:
Got commented on this on a sub-branch, so it's already modified. Applied those changes here as well!